### PR TITLE
refactor: CLI commands and helpers

### PR DESCRIPTION
**Pull request by uchilaka**
## Ticket(s)

- [LAR-265](https://linear.app/larcity-and-affiliates/issue/LAR-265)

## Summary

This PR focuses on refactoring CLI commands and helpers to improve efficiency, error handling, and developer experience. It also includes critical asset manifest fixes and dependency cleanup.

### Implementation Plan
- **CLI Refactoring:** Improved `DDNSCmd` record fetching logic to be more memory-efficient and added robust handling for cases with no or multiple records.
- **Error Handling:** Updated `ServicesCmd` to use `Thor::Error` for cleaner CLI exit messages.
- **Helper Enhancements:** Extended `OutputHelpers` with customizable warning prefixes and added `mock_return` support to the `run` helper in `Runnable` for better testability.
- **Asset Optimization:** Fixed a duplicate `actiontext.css` link in `manifest.js` and removed the redundant `sassc-rails` gem (superseded by `dartsass-rails`).
- **Validation:** Added and verified test coverage for core CLI commands and helper modules.

### /sitrep (Situation Report)
- **Current State:** The branch is fully implemented and all relevant CLI tests are passing in the `test` environment.
- **Accomplished:** We've streamlined the DDNS management logic and hardened the CLI toolset against common edge cases. The asset pipeline is now cleaner with the removal of duplicate links and legacy gems.
- **Outstanding Items:**
  - A few pending tests in `DDNSCmd` are marked for future evaluation of "sloppy" test logic.
  - An unresolved TODO in `runnable.rb` regarding `eval` conditions remains for future review.

## Screenshots (if appropriate)

N/A

## Review app(s)

N/A

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the README/confluence documentation. (Fixed manifest)
- [x] I have updated the relevant documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

## How to review

Please review the changes in `lib/commands/lar_city/cli/ddns_cmd.rb` and `lib/lar_city/cli/runnable.rb`. Verify that the `Thor::Error` usage in `ServicesCmd` provides the expected clean output.

- [x] Code looks chill? 🚀

### CODE REVIEW

File: lib/commands/lar_city/cli/ddns_cmd.rb [hunk 1]
```diff
@@ -87,6 +96,64 @@ def cleanup
...
+            per_page = [batch_size, 100].min
+            page = 1
+            # Iterate over the pages and fetch records as long as a result is returned
+            while (next_records = get_records(domain:, name:, type: record_type, access_token:, page:, per_page:))&.any?
+              # Exit while loop if we exceed the batch size
+              break if records.size >= batch_size
+
+              records += next_records
+              # say_info "Found #{records.size} records for #{name} on #{domain} (page #{page})"
+              page += 1
```
[the remark this answers is not in the document]
Comment (uchilaka) [reply lib/commands/lar_city/cli/ddns_cmd.rb]: This implementation suggestion seems stale
